tests/assets/test_divisible_assets.py

```diff
@@ -0,0 +1,686 @@
+    b.write_block(block, durability='hard')
+    # vote
+    vote = b.vote(block.id, b.get_last_voted_block().id, True)
+    b.write_vote(vote)
+
+    # TRANSFER
+    # `user_vk` combines two different transaction with 50 shares each and
+    # transfers a total of 100 shares back to `b`
+    tx_transfer2 = Transaction.transfer([tx_create.to_inputs()[0],
+                                         tx_transfer1.to_inputs()[0]],
+                                        [([b.me], 100)],
+                                        asset=tx_create.asset)
+    tx_transfer2_signed = tx_transfer2.sign([user_sk])
+
+    assert tx_transfer2_signed.validate(b) == tx_transfer2_signed
+    assert len(tx_transfer2_signed.conditions) == 1
+    assert tx_transfer2_signed.conditions[0].amount == 100
+    assert len(tx_transfer2_signed.fulfillments) == 2
+
+    fid0_input = tx_transfer2_signed.fulfillments[0].to_dict()['input']['txid']
+    fid1_input = tx_transfer2_signed.fulfillments[1].to_dict()['input']['txid']
+    assert fid0_input == tx_create.id
+    assert fid1_input == tx_transfer1.id
+
+
+# In a TRANSFER transaction of a divisible asset the amount being spent in the
+# inputs needs to match the amount being sent in the outputs.
+# In other words `amount_in_inputs - amount_in_outputs == 0`
+@pytest.mark.usefixtures('inputs')
+def test_amount_error_transfer(b, user_vk, user_sk):
+    from bigchaindb.models import Transaction
+    from bigchaindb.common.transaction import Asset
+    from bigchaindb.common.exceptions import AmountError
+
+    # CREATE divisible asset
+    asset = Asset(divisible=True)
+    tx_create = Transaction.create([b.me], [([user_vk], 100)], asset=asset)
+    tx_create_signed = tx_create.sign([b.me_private])
+    # create block
+    block = b.create_block([tx_create_signed])
+    assert block.validate(b) == block
+    b.write_block(block, durability='hard')
+    # vote
+    vote = b.vote(block.id, b.get_last_voted_block().id, True)
+    b.write_vote(vote)
+
+    # TRANSFER
+    # output amount less than input amount
+    tx_transfer = Transaction.transfer(tx_create.to_inputs(), [([b.me], 50)],
+                                       asset=tx_create.asset)
+    tx_transfer_signed = tx_transfer.sign([user_sk])
+    with pytest.raises(AmountError):
+        tx_transfer_signed.validate(b)
+
+    # TRANSFER
+    # output amount greater than input amount
+    tx_transfer = Transaction.transfer(tx_create.to_inputs(), [([b.me], 101)],
+                                       asset=tx_create.asset)
+    tx_transfer_signed = tx_transfer.sign([user_sk])
+    with pytest.raises(AmountError):
+        tx_transfer_signed.validate(b)
+
+
+@pytest.mark.skip(reason='Figure out how to handle this case')
+@pytest.mark.usefixtures('inputs')
+def test_threshold_same_public_key(b, user_vk, user_sk):
+    # If we try to fulfill a threshold condition where each subcondition has
+    # the same key get_subcondition_from_vk will always return the first
```

Comment:
I think the problem comes from https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/common/transaction.py#L969

This only happens when you create a threshold condition with the same public key which makes no sense in the first place. Example if owners_after = `[user_pub, user_pub, user_pub]` you end up with 3 subconditions for the same public key but when you do `subffill = ccffill.get_subcondition_from_vk(owner_before)[0]` you end up always fulfilling the first subcondition